src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -3,30 +3,58 @@
 
 using System.Collections.Generic;
 using System.Diagnostics;
-using System.Linq;
+using System.Reflection;
 using System.Text.Json.Serialization;
-using System.Text.Json.SourceGeneration.Reflection;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Text;
 
 namespace System.Text.Json.SourceGeneration
 {
     public sealed partial class JsonSourceGenerator
     {
-        private sealed class Emitter
+        private sealed partial class Emitter
         {
+            // Literals in generated source
             private const string RuntimeCustomConverterFetchingMethodName = "GetRuntimeProvidedCustomConverter";
-
-            private const string JsonContextDeclarationSource = "internal partial class JsonContext : JsonSerializerContext";
-
             private const string OptionsInstanceVariableName = "Options";
-
-            private const string PropInitFuncVarName = "PropInitFunc";
-
-            private const string JsonMetadataServicesClassName = "JsonMetadataServices";
-
+            private const string PropInitMethodNameSuffix = "PropInit";
+            private const string SerializeMethodNameSuffix = "Serialize";
             private const string CreateValueInfoMethodName = "CreateValueInfo";
-
+            private const string DefaultOptionsStaticVarName = "s_defaultOptions";
```

Comment:
I wonder if we can use `[InternalsVisibleTo]` in these cases to avoid reflection.

Reply:
Where is this used with reflection?

Reply:
Sorry, assumed reflection. Looks like it's just for generation.

Reply:
There's no reflection (in the generated source, tests, or elsewhere). These constants are for values (e.g. variable names) that appear in the generated source.